`process.stdout` is undefined in many cases on Windows OS and Vitest uses it in few places

### Describe the bug

`process.stdout` is often seem to be causing issues on Windows and is often undefined in many cases on Windows OS and Vitest is trying to get TTY width in a few areas by calling `process.stdout.columns` (and maybe `rows` too) and this throws an error in Windows. This bug is very similar to another bug #1231 that I opened that got fixed not long ago for the same reason. So as far as I can see `process.stdout` really seems to be a problem on Windows. 

I also found this SO asking similar question: 
[SO - How can I access stdout.columns in Node on Windows?](https://stackoverflow.com/questions/19062906/how-can-i-access-stdout-columns-in-node-on-windows)

and other GitHub projects with similar issue
[Facebook Flow - Support for `process.stdout.columns`](https://github.com/facebook/flow/issues/4487)
- from this issue, it seems that `process.stdout.columns` (and `rows`) is only available when `process.stdout.isTTY` is `true`

Searching for `process.stdout` in Vitest code is returning 7 places, 1 of them got fixed on Windows by calling `console._stdout` as mentioned in #1231 and fixed in #1235. Considering this, would it be possible to find alternative to `process.stdout` to avoid Vitest itself throwing errors on Windows platform?

### Reproduction
This simple unit test on Windows will throw a Vitest error coming from `process.stdout.columns` (where `process.stdout` is undefined and throws that it cannot read "columns" from undefined)
```ts
it('should test a spy', async () => {
    const obj = {
      getUsers: () => Promise.resolve({ id: 123, name: 'John' }),
    };

    const spy = vi.spyOn(obj, 'getUsers');

    expect(await obj.getUsers()).toEqual({ id: 123, name: 'John' });
    expect(spy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledWith({ id: 999, name: 'John' });
  });
```
will throw this Vitest error
```shell
 FAIL  src/stores/__tests__/userStore.spec.ts > User Store > should test a spy
TypeError: Cannot read properties of undefined (reading 'columns')
 ❯ src/stores/__tests__/userStore.spec.ts:46:16
     44|     expect(await obj.getUsers()).toEqual({ id: 123, name: 'John' });
     45|     expect(spy).toHaveBeenCalled();
     46|     expect(spy).toHaveBeenCalledWith({ id: 999, name: 'John' });
       |                ^
     47|   });
     48| });
```

if I go in Vitest internal code and change `process.stdout.columns` to `process.stdout && process.stdout.columns` then it works and I see the full error
```shell
 FAIL  src/stores/__tests__/userStore.spec.ts > User Store > should test a spy
AssertionError: expected "getUsers" to be called with arguments: [ { id: 999, name: 'John' } ]expected "getUsers" to be called with arguments: [ { id: 999, name: 'John' } ]

Received:
    1st getUsers call:

  - Array [
  -   Object {
  -     "id": 999,
  -     "name": "John",
  -   },
  - ]
  + Array []


Number of calls: 1

 ❯ src/stores/__tests__/userStore.spec.ts:46:16
     44|     expect(await obj.getUsers()).toEqual({ id: 123, name: 'John' });
     45|     expect(spy).toHaveBeenCalled();
     46|     expect(spy).toHaveBeenCalledWith({ id: 999, name: 'John' });
       |                ^
     47|   });
     48| });
```

For my use case, the error seems to come from `formatLine`
https://github.com/vitest-dev/vitest/blob/b027d1617eac9da698255a0a7ae3e1fe3a54e872/packages/vitest/src/node/diff.ts#L5-L7

changing it to this line no longer throws from Vitest and allows me to see the real error
```ts
return cliTruncate(line, (outputTruncateLength ?? (process.stdout && process.stdout.columns || 80)) - 4)
```
Again this is really similar to the error I had reported earlier in #1231

### System Info

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 13.19 GB / 31.75 GB
  Binaries:
    Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.15 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 100.0.4896.127
    Edge: Spartan (44.19041.1266.0), Chromium (101.0.1210.32)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    vite: ^2.9.7 => 2.9.5
    vitest: workspace:* => 0.10.2
```


### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vitest/src/node/core.ts
+++ b/packages/vitest/src/node/core.ts
@@ -283,7 +283,7 @@ export class Vitest {
     if (this.server.config.clearScreen === false)
       return
 
-    const repeatCount = process.stdout.rows - 2
+    const repeatCount = (process.stdout?.rows ?? 0) - 2
     const blank = repeatCount > 0 ? '\n'.repeat(repeatCount) : ''
     this.console.log(blank)
     readline.cursorTo(process.stdout, 0, 0)

--- a/packages/vitest/src/node/diff.ts
+++ b/packages/vitest/src/node/diff.ts
@@ -3,7 +3,7 @@ import * as diff from 'diff'
 import cliTruncate from 'cli-truncate'
 
 export function formatLine(line: string, outputTruncateLength?: number) {
-  return cliTruncate(line, (outputTruncateLength ?? (process.stdout.columns || 80)) - 4)
+  return cliTruncate(line, (outputTruncateLength ?? (process.stdout?.columns || 80)) - 4)
 }
 
 export interface DiffOptions {

--- a/packages/vitest/src/node/error.ts
+++ b/packages/vitest/src/node/error.ts
@@ -189,7 +189,7 @@ export function generateCodeFrame(
   let count = 0
   let res: string[] = []
 
-  const columns = process.stdout.columns || 80
+  const columns = process.stdout?.columns || 80
 
   function lineNo(no: number | string = '') {
     return c.gray(`${String(no).padStart(3, ' ')}| `)

--- a/packages/vitest/src/node/reporters/renderers/utils.ts
+++ b/packages/vitest/src/node/reporters/renderers/utils.ts
@@ -12,7 +12,7 @@ export const pointer = c.yellow(F_POINTER)
 export const skipped = c.dim(c.gray(F_DOWN))
 
 export function getCols(delta = 0) {
-  let length = process.stdout.columns
+  let length = process.stdout?.columns
   if (!length || isNaN(length))
     length = 30
   return Math.max(length + delta, 0)

